File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-0.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        rowZero = False
        m,n = len(matrix), len(matrix[0])

        for i in range(m):
            for j in range(n):
                if matrix[i][j]==0:
                    matrix[0][j]=0
                    if i>0:
                        matrix[i][0]=0
                    if i==0:
                        rowZero = True
        
        
        for r in range(1,m):
            for c in range(1, n):
                if matrix[r][0]==0 or matrix[0][c]==0:
                    matrix[r][c]=0

        if matrix[0][0]==0:
            for i in range(m):
                matrix[i][0]=0
        
        if rowZero==True:
            for j in range(n):
                matrix[0][j]=0   

                

File: Data Structures & Algorithms/search-2d-matrix/submission-0.js
class Solution {
    /**
     * @param {number[][]} matrix
     * @param {number} target
     * @return {boolean}
     */
    searchMatrix(matrix, target) {
        const m = matrix.length
        const n = matrix[0].length
        let low = 0, high = m*n-1
        while(low<=high){
            const mid = Math.floor((low+high)/2)
            const row = Math.floor(mid/n)
            const col = mid%n
            if(matrix[row][col] < target){
                low = mid+1
            } else if(matrix[row][col] > target){
                high = mid-1
            } else{
                return true
            }
        }
        return false
    }
}
